symbolic-debuginfo/src/breakpad.rs

```diff
@@ -614,6 +644,69 @@ impl<'d> Iterator for BreakpadLineRecords<'d> {
     }
 }
 
+/// An [inline record] associated with a `BreakpadFunctionRecord`.
+///
+/// Inline records are so frequent in a Breakpad symbol file that they do not have a record
+/// identifier. They immediately follow the [`BreakpadFuncRecord`] that they belong to. Thus, an
+/// iterator over inline records can be obtained from the function record.
+///
+/// Example: `INLINE 1 61 1 2 7b60 3b4`
+///
+/// [inline record]: https://github.com/google/breakpad/blob/main/docs/symbol_files.md#inline-records
+/// [`BreakpadFuncRecord`]: struct.BreakpadFuncRecord.html
+#[derive(Clone, Debug, Default, Eq, PartialEq)]
+pub struct BreakpadInlineRecord {
+    /// The depth of nested inline calls.
+    pub inline_depth: u64,
+    /// The line number of the call, in the parent function. Zero means no line number.
+    pub call_site_line: u64,
+    /// Identifier of the [`BreakpadFileRecord`] specifying the file name of the line of the call.
+    pub call_site_file_id: u64,
+    /// Identifier of the [`BreakpadInlineOriginRecord`] specifying the function name.
+    pub origin_id: u64,
+    /// A list of address ranges which contain the instructions for this inline call. Contains at
+    /// least one element.
+    pub address_ranges: Vec<BreakpadInlineAddressRange>,
+}
+
+impl BreakpadInlineRecord {
+    /// Parses a line record from a single line.
+    pub fn parse(data: &[u8]) -> Result<Self, BreakpadError> {
+        let string = str::from_utf8(data)?;
+        Ok(parsing::inline_record_final(string.trim())?)
+    }
+
+    /// Resolves the filename for this record in the file map.
+    pub fn call_site_filename<'d>(&self, file_map: &BreakpadFileMap<'d>) -> Option<&'d str> {
+        file_map.get(&self.call_site_file_id).cloned()
+    }
+
+    /// Resolves the function name for this record in the inline origin map.
+    pub fn function_name<'d>(
+        &self,
+        inline_origin_map: &BreakpadInlineOriginMap<'d>,
+    ) -> Option<&'d str> {
+        inline_origin_map.get(&self.origin_id).cloned()
+    }
```

Comment:
Honestly, I would inline these functions. They are each only used once.

Reply:
Sounds good. I created `BreakpadInlineRecord` by copying `BreakpadLineRecord` which already has a similar method for the `filename`, but I agree there's not much value in having those methods.